Data/ByteString/Lazy.hs

```diff
@@ -1000,7 +1000,7 @@ findIndexEnd k = findIndexEnd' 0
     findIndexEnd' _ Empty = Nothing
     findIndexEnd' n (Chunk c cs) =
       let !n' = n + S.length c
-          !i  = fmap (fromIntegral . (n +)) $ S.findIndexEnd k c
+          !i  = fromIntegral . (n +) <$> S.findIndexEnd k c
```

Comment:
Because it requires CPP-guarded `import Control.Applicative` and I was lazy to add it :) Fixed now. 

(I suggest an upcoming `bytestring-0.11.1.0` to be the last release supporting GHC 7.0, because it becomes really annoying to workaround AMP/SMP all the time)

Reply:
> (I suggest an upcoming `bytestring-0.11.1.0` to be the last release supporting GHC < 8.0, because it becomes really annoying to workaround AMP/SMP all the time)

Sounds good to me.